minijinja/src/value/mod.rs

```diff
@@ -583,23 +593,47 @@ impl Value {
         }
     }
 
+    /// If the value is a sequence it's returned as [`SeqObject`].
+    pub fn as_seq(&self) -> Option<&dyn SeqObject> {
+        match self.0 {
+            ValueRepr::Seq(ref v) => return Some(&**v as &dyn SeqObject),
+            ValueRepr::Dynamic(ref dy) => {
+                if let ObjectKind::Seq(seq) = dy.kind() {
+                    return Some(seq);
+                }
+            }
+            _ => {}
+        }
+        None
+    }
+
+    /// If the value is a struct, return it as [`StructObject`].
+    pub fn as_struct(&self) -> Option<&dyn StructObject> {
```

Comment:
Should this also return `Some()` when we have a `ValueRepr::Map`? That would make its functionality similar to `as_seq()`. And if so, maybe `StructObject` should be called `MapObject`.

Reply:
At a later point I might introduce a `MapObject` which takes arbitrary keys. However that opens up some uncomfortable questions about how to represent keys which I don't want to go into yet.